pkgs/data/misc/cacert/default.nix

```
@@ -1,123 +1,211 @@
 { lib
 , stdenv
 , fetchurl
-, nss
-, python3
-, blacklist ? [ ]
+, buildcatrust
+, blacklist ? []
+, extraCertificates ? []
 
-  # Used for tests only
+# Used for overriding certdata version (see update.sh)
+, nssOverride ? null
+
+# Used for tests only
 , runCommand
 , cacert
 , openssl
 }:
 
+# extraCertificates should be paths or strings.
+assert lib.length (lib.filter (f: !(builtins.isPath f || builtins.isString f)) extraCertificates) == 0;
+
 let
-  certdata2pem = fetchurl {
-    name = "certdata2pem.py";
-    urls = [
-      "https://salsa.debian.org/debian/ca-certificates/raw/debian/20170717/mozilla/certdata2pem.py"
-      "https://git.launchpad.net/ubuntu/+source/ca-certificates/plain/mozilla/certdata2pem.py?id=47e49e1e0a8a1ca74deda27f88fe181191562957"
-    ];
-    sha256 = "1d4q27j1gss0186a5m8bs5dk786w07ccyq0qi6xmd2zr1a8q16wy";
+  isPathIntoStore = f: builtins.isString f && lib.substring 0 (lib.stringLength builtins.storeDir) f == builtins.storeDir;
+  extraCertificateFiles = lib.filter (f: builtins.isPath f || isPathIntoStore f) extraCertificates;
+  extraCertificateStrings = lib.filter builtins.isString extraCertificates;
```

I think this needs something like this:

```suggestion
  extraCertificateStrings = lib.filter (f: builtins.isString f && !(lib.isStorePath f)) extraCertificates;
```

Otherwise we'll mistake values like `"${foo}/bar.crt` for certificate strings.